src/doc/trpl/guessing-game.md

```diff
@@ -0,0 +1,991 @@
+% Guessing Game
+
+For our first project, we’ll implement a classic beginner programming problem:
+the guessing game. Here’s how it works: Our program will generate a random
+integer between one and a hundred. It will then prompt us to enter a guess.
+Upon entering our guess, it will tell us if we’re too low or too high. Once we
+guess correctly, it will congratulate us. Sound good?
+
+# Set up
+
+Let’s set up a new project. Go to your projects directory. Remember how we had
+to create our directory structure and a `Cargo.toml` for `hello_world`? Cargo
+has a command that does that for us. Let’s give it a shot:
+
+```bash
+$ cd ~/projects
+$ cargo new guessing_game --bin
+$ cd guessing_game
+```
+
+We pass the name of our project to `cargo new`, and then the `--bin` flag,
+since we’re making a binary, rather than a library.
+
+Check out the generated `Cargo.toml`:
+
+```toml
+[package]
+
+name = "guessing_game"
+version = "0.0.1"
+authors = ["Your Name <you@example.com>"]
+```
+
+Cargo gets this information from your environment. If it’s not correct, go ahead
+and fix that.
+
+Finally, Cargo generated a ‘Hello, world!’ for us. Check out `src/main.rs`:
+
+```rust
+fn main() {
+    println!("Hello, world!")
+}
+```
+
+Let’s try compiling what Cargo gave us:
+
+```{bash}
+$ cargo build
+   Compiling guessing_game v0.0.1 (file:///home/you/projects/guessing_game)
+```
+
+Excellent! Open up your `src/main.rs` again. We’ll be writing all of
+our code in this file.
+
+Before we move on, let me show you one more Cargo command: `run`. `cargo run`
+is kind of like `cargo build`, but it also then runs the produced executable.
+Try it out:
+
+```bash
+$ cargo run
+   Compiling guessing_game v0.0.1 (file:///home/you/projects/guessing_game)
+     Running `target/debug/guessing_game`
+Hello, world!
+```
+
+Great! The `run` command comes in handy when you need to rapidly iterate on a
+project. Our game is just such a project, we need to quickly test each
+iteration before moving on to the next one.
+
+# Processing a Guess
+
+Let’s get to it! The first thing we need to do for our guessing game is
+allow our player to input a guess. Put this in your `src/main.rs`:
+
+```rust,no_run
+use std::io;
+
+fn main() {
+    println!("Guess the number!");
+
+    println!("Please input your guess.");
+
+    let mut guess = String::new();
+
+    let input = io::stdin().read_line(&mut guess)
+        .ok()
+        .expect("Failed to read line");
```

Comment:
yeah, it's debateable. I think i'll leave it like this for now, though. This is, of course, ridiculously subjective.